zapcore/field.go

```diff
@@ -102,11 +102,12 @@ const (
 // context. Most fields are lazily marshaled, so it's inexpensive to add fields
 // to disabled debug-level log statements.
 type Field struct {
-	Key       string
-	Type      FieldType
-	Integer   int64
-	String    string
-	Interface interface{}
+	Key            string
+	Type           FieldType
+	Integer        int64
+	String         string
+	Interface      interface{}
+	DisableErrorVerbose bool
```

Comment:
We avoid adding new struct fields to the `Field` type for performance reasons, as the struct is passed as a value.

We could add a new `FieldType` to indicate an error that should only have the message included, though that could have backwards compatibility concerns with some encoders.

Reply:
@prashantv 

I appreciate your feedback. I will create a new type with error field and DisableErrorVerbose field.